tasks/winbuild.py

```diff
@@ -0,0 +1,70 @@
+import os
+import shutil
+
+from invoke.tasks import task
+
+from tasks.flavor import AgentFlavor
+from tasks.libs.common.utils import get_version
+from tasks.msi import build as agent_msi_build
+from tasks.msi import build_installer
+from tasks.omnibus import build as omnibus_build
+
+# Output directory for package files
+OUTPUT_PATH = os.path.join(os.getcwd(), "omnibus", "pkg")
+# Omnibus stores files here, e.g. C:\opt\datadog-agent, C:\opt\dataog-installer
+OPT_SOURCE_DIR = os.path.join('C:\\', 'opt')
+
+
+@task
+def agent_package(
```

Comment:
:suggestion: I think it was not advised to call tasks from another task (but it was maybe a namespace problem, I'm not 100% sure). What do you think of calling explicitly these tasks from the powershell script?

Reply:
> I think it was not advised to call tasks from another task (but it was maybe a namespace problem, I'm not 100% sure)

I haven't been able to find a source for that. The only oddity I can see is just a plain Python thing. Many of our invoke tasks and helperfunctions define defaults in the function definition, if the defaults change in an inner function then any out outer calls will be using the wrong default. I typically prefer to use `None` as the default so outers can defer to the inner call for default values.
```python
def inner(release_version=None):
    if release_version is None:
        release_version = 'nightly-a7'
def outer(release_version=None):
    inner(release_version=release_version)
```

> What do you think of calling explicitly these tasks from the powershell script?

hmmm, perhaps. I think it's up to if people would rather modify PowerShell to add additional build steps, or if they'd rather modify Python. I like having the PowerShell script handle the env related things that can't be done in Python and have Python handle the actual "build".

> I think that would be cool but how would developers do all the above in a single step?

User would run `Build-AgentPacakge.ps1`. Though if your env is already properly configured you could also just run the invoke task.